drivers/i2s/i2s_nrfx.c

```diff
@@ -345,6 +366,7 @@ static void data_handler(const struct device *dev,
 				 * before this buffer would be started again).
 				 */
 				next.p_tx_buffer = drv_data->last_tx_buffer;
+				next.buffer_size = drv_data->last_buffer_size;
```

Comment:
You could just use `next.buffer_size = 1` here as this block will not be transferred anyway (see the comment above). Similarly in the `I2S_STATE_STOPPING` state below. It would not be necessary to store `last_buffer_size` then.